lib/core/versions/latest/Did.ts

```diff
@@ -62,19 +62,43 @@ export default class Did {
    * @param didString Short or long-form DID string.
    */
   public static async create (didString: string, didMethodName: string): Promise<Did> {
+    const methodNameParts = didMethodName.split(':');
+    if (methodNameParts.length < 2) {
+      throw new SidetreeError(ErrorCode.DidInvalidMethodName);
+    }
+
     const did = new Did(didString, didMethodName);
 
     // If DID is long-form, ensure the unique suffix constructed from the suffix data matches the short-form DID and populate the `createOperation` property.
     if (!did.isShortForm) {
-      const indexOfSemiColonChar = didString.indexOf(';');
-      const didParameterString = didString.substring(indexOfSemiColonChar + 1);
-
-      if (!didParameterString.startsWith(Did.initialValuesParameterPrefix)) {
-        throw new SidetreeError(ErrorCode.DidLongFormOnlyInitialValuesParameterIsAllowed);
+      let didStringUrl = undefined;
+      try {
+        didStringUrl = new URL(didString);
+      } catch {
+        throw new SidetreeError(ErrorCode.DidInvalidDidString);
+      }
+      let queryParamCounter = 0;
+      let encodedCreateRequest = '';
+
+      didStringUrl.searchParams.forEach((value, key) => {
+        queryParamCounter += 1;
+        if (queryParamCounter > 1) {
+          throw new SidetreeError(ErrorCode.DidLongFormOnlyInitialStateParameterIsAllowed);
+        }
+
+        // expect key to be -<method>-initial-state
+        const expectedKey = `-${methodNameParts[1]}-${this.initialStateParameterSuffix}`;
+        if (key !== expectedKey) {
+          throw new SidetreeError(ErrorCode.DidLongFormOnlyInitialStateParameterIsAllowed);
```

Comment:
Nit:
As a rule of thumb, we try never reuse error code so that there is no ambiguity which line through the error. Consider creating dedicated error code for each.

Reply:
Or maybe include the incorrect key in the error code.

```suggestion
          throw new SidetreeError(ErrorCode.DidLongFormOnlyInitialStateParameterIsAllowed, `Incorrect parameter: ${key}`);
```